syntax = "proto3";

import "validate.proto";

enum LogId
{
    UNKNOWN     = 0;        //no message corresponding to this value
    VIDEO_START = 100;      //%%#message:: VideoStart,VIDEO_START
    VIDEO_END   = 101;      //%%#message:: VideoEnd,VIDEO_END
}


message LogHdr {
    LogId    log_id = 1;
}

message VideoStart
{
    LogId       log_id = 1;
    string      s_pl_cookie = 3;
    uint32      u_segment_id = 4;
    uint32      u_duration_ms = 5;
}

message Cookie {
    string      cookie = 1 [(validate.rules).string = {in: ["123", "234", "456"]}];
}

message VideoEnd
{
    LogId       log_id = 1;
    string      ev_name = 2 [(validate.rules).string = {in: ["foo", "bar", "baz"]}];
    string      s_pl_cookie = 3;
    uint32      u_segment_id = 4;
    uint32      u_played_dur = 5;
    Cookie      tmp_cookie = 6;
}


